scripts/install.sh

```diff
@@ -0,0 +1,190 @@
+#!/bin/bash -e
+#
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+if [ "$(id -u)" -ne 0 ]; then
+    echo "ERROR: please run this script as root"
+    exit 1
+fi
+
+if ! snap list | grep -q anbox-cloud-appliance; then
+  echo "ERROR: Anbox Cloud Appliance not installed"
+  exit 1
+fi
+
+LOCAL_SNAP=
+SERVICE_NAME=cloud-gaming-demo
+DEMO_SNAP_COMMON_DIR="/var/snap/cloud-gaming-demo/common"
+APPLIANCE_SNAP_COMMON_DIR="/var/snap/anbox-cloud-appliance/common"
+
+GAMES="minetest bombsquad mindustry"
+MINETEST_DOWNLOAD_URL="https://github.com/minetest/minetest/releases/download/5.5.0/app-arm64-v8a-release.apk"
+MINETEST_PKG_ARCH="arm64"
+BOMBSQUAD_DOWNLOAD_URL="https://files.ballistica.net/bombsquad/builds/BombSquad_Android_Generic_1.6.10.apk"
+BOMBSQUAD_PKG_ARCH="universal"
+MINDUSTRY_DOWNLOAD_URL="https://f-droid.org/repo/io.anuke.mindustry_104.apk"
+MINDUSTRY_PKG_ARCH="universal"
+
+print_help() {
+    echo "Usage: ${0} [OPTIONS]"
+    echo "       install and configure cloud gaming demo to make it work against an Anbox Cloud Appliance deployment"
+    echo
+    echo "arguments:"
+    echo " --local-snap=<string>             Path to the local snap for cloud gaming demo installation"
+    echo " -h|--help                         Show help"
+}
+
+
+while [ -n "$1" ]; do
+    case "$1" in
+        --local-snap=*)
+            LOCAL_SNAP=${1#*=}
+            shift
+            ;;
+        -h|--help)
+            print_help
+            exit 2
+            ;;
+        *)
+            echo "ERROR: Unsupported argument: $1"
+            exit 1
+            ;;
+    esac
+done
+
+generate_config_file() {
+  local gateway_address="$1"
+  # Due to a bug(AC-777), use the juju command to create a gateway token
+  # rather than using anbox-cloud-appliance.
+  local token=$(sudo -u ubuntu juju ssh -m appliance:anbox-cloud anbox-stream-gateway/0 \
+                -- "sudo /snap/bin/anbox-stream-gateway account create cloud-gaming-demo-$(date +%s)")
+
+  local service_folder="${DEMO_SNAP_COMMON_DIR}/service"
+  mkdir -p "${service_folder}" && chmod 0750 "${service_folder}"
+  cat << EOF > "${service_folder}/config.yaml"
+gateway-url: https://${gateway_address}
+gateway-token: ${token}
+EOF
+
+  chmod -R 0600 "${service_folder}/config.yaml"
+}
+
+install_snap() {
+  if snap list | grep cloud-gaming-demo ; then
+    snap remove --purge cloud-gaming-demo
+  fi
+
+  if [ -n $LOCAL_SNAP ];  then
```

Comment:
This needs quotes, otherwise the check will fail
```
ubuntu@kalmar:~/morphis$ sudo ./install.sh 
Install cloud gaming demo from the local snap...
error: cannot install snap with empty name
```

Reply:
+1, will fix